paint/PAINT_ShapeTool.cpp

```diff
@@ -1,16 +1,51 @@
 #include "PAINT_pch.h"
 #include "PAINT_ShapeTool.h"
+#include "PAINT_Brush.h"
+#include "GFX_Rectangle.h"
 
 using namespace paint;
 
-ShapeTool::ShapeTool()
+ShapeTool::ShapeTool(SDL_Renderer* renderer, SDL_Texture* texture)
 	: activeShape_(nullptr)
+	, renderer_(renderer)
+	, texture_(texture)
 {
-	
+	activeBrush_ = std::make_shared<Brush>(0);
```

Comment:
No protected fields, please! Add a constructor with a brush parameter if required, or default the brush in the defining base class to make one of size 0.
**MUST**

Reply:
done